ggshield/iac/output/iac_text_output_handler.py

```diff
@@ -169,3 +345,33 @@ def no_iac_vulnerabilities() -> str:
     Build a message if no IaC vulnerabilities were found.
     """
     return format_text("\nNo incidents have been found\n", STYLE["no_secret"])
+
+
+def diff_scan_summary(
+    new: List[IaCFileResult],
+    unchanged: List[IaCFileResult],
+    deleted: List[IaCFileResult],
+) -> str:
+    def detail(entries: List[IaCFileResult]) -> str:
+        count: Dict[str, int] = dict()
+        for entry in entries:
+            for incident in entry.incidents:
+                count.setdefault(incident.severity, 0)
+                count[incident.severity] += 1
+        formatted_count = [f"{key}: {val}" for key, val in count.items()]
+        if len(formatted_count) == 0:
+            return ""
+        return f" ({', '.join(formatted_count)})"
+
+    buf = StringIO()
+    buf.write("\nSummary of changes:\n")
+    buf.write(
+        f'[-] {len(deleted)} {pluralize("incident", len(deleted), "incidents")} deleted{detail(deleted)}\n'
+    )
+    buf.write(
+        f'[~] {len(unchanged)} {pluralize("incident", len(unchanged), "incidents")} remaining{detail(unchanged)}\n'
+    )
+    buf.write(
+        f'[+] {len(new)} new {pluralize("incident", len(new), "incidents")} detected{detail(new)}\n'
```

Comment:
We use a lot `pluralize("incident", number, "incidents")`, could we factorize this ?